Cycle 3 addressed in d28f7590f.

**1. `parseRoundingMode` Javadoc contradiction - correct catch, fixed.** You are right: `args.length == 3 ? parseRoundingMode(args[2]) : HALF_UP` ran *before* the propagation check, so an explicit null was parsed to HALF_UP and the result then discarded. The behaviour was harmless but the Javadoc's claim that an explicit null "never reaches this method" was simply false.

Fixed by short-circuiting rather than by softening the comment, so the contract becomes true instead of merely accurate:

```java
final RoundingMode mode =
    args.length == 3 && !CypherFunctionHelper.isExplicitNull(args, 2) ? parseRoundingMode(args[2]) : RoundingMode.HALF_UP;
```

The ordering guarantee is unaffected - a mode that is present and non-null is still parsed before propagation, so `round(null, 2, 'SIDEWAYS')` keeps reporting the unusable mode. `anUnusableModeIsStillReportedWhenAnotherArgumentIsNull` still passes. The Javadoc now also records that the `null` branch serves callers passing the omitted-argument sentinel directly, which is what keeps it from being dead.

**2. Truncate computes before short-circuiting - declining, and not only on cost grounds.** I tried the suggested `&& !isExplicitNull` on the earlier line to be sure, and it **breaks the ordering guarantee**:

```
[ERROR] CypherOptionalArgumentNullIssue5629Test.aBadTruncateUnitIsStillReportedWhenTheAdjustmentMapIsNull:191
```

Skipping `TemporalUtil.truncate*` means `date.truncate('fortnight', d, null)` answers `null` instead of reporting the bad unit - exactly the masking this PR was careful to avoid. The wasted computation is the price of validating first, and it is one truncation on a path that is about to return `null` anyway. Reverted; the probe is not in the branch.

This is the difference between the two cases: in `round()` the parse is a *pure* function of the argument being discarded, so skipping it loses nothing. In `truncate()` the computation is what performs the validation, so skipping it loses the error.

**3. Follow-up #1 priority - agreed.** Noted for a fast-follow; it is the raw `NullPointerException` on a null unit plus the `IllegalArgumentException` on an unknown one, both HTTP 500 for a caller mistake.

--- a/docs/5629-cypher-explicit-null-optional-argument.md
+++ b/docs/5629-cypher-explicit-null-optional-argument.md
@@ -187,3 +187,12 @@ whoever cuts the release should carry it into the release notes.
 5. **`com.arcadedb.function.text.SubstringFunction` is registered by no factory**, so Cypher's `substring()` and
    this class are two implementations of one function with only one of them reachable. Either wire it up or
    delete it; keeping both invites the divergence this change just repaired.
+6. **The `*.truncate` family silently ignores a present, non-null, non-map third argument.**
+   `date.truncate('year', d, 42)` answers the truncated date and drops the `42`, because the `args[2] instanceof
+   Map` guard is the only thing looking at it. That is the same shape of defect this change repaired for `null`,
+   one step further out: a wrong query looks like a successful one. It should be a type error.
+
+   Deliberately **not** covered by a characterization test here. Asserting that `date.truncate('year', d, 42)`
+   returns the truncated value would encode the defect as the expected behaviour and make fixing it look like a
+   regression - the same trap avoided with the unknown-unit exception class above. It belongs with follow-up 1,
+   as one pass over the family's argument handling.

--- a/engine/src/main/java/com/arcadedb/function/math/RoundFunction.java
+++ b/engine/src/main/java/com/arcadedb/function/math/RoundFunction.java
@@ -63,16 +63,14 @@ public class RoundFunction implements StatelessFunction {
     // parse-time check applies, which examines each argument independently (issue #5484).
     final Number number = CypherFunctionHelper.requireNumberArgument(args[0], "round");
     final Number precisionArg = args.length > 1 ? CypherFunctionHelper.requireNumberArgument(args[1], "round") : null;
-    // An explicitly written null mode is not parsed: it propagates below, so parsing it only to discard the HALF_UP it
-    // would yield would also make parseRoundingMode's contract untrue. A mode that is present and not null is still
-    // parsed here, before propagation, so round(null, 2, 'SIDEWAYS') reports the unusable mode.
-    final RoundingMode mode =
-        args.length == 3 && !CypherFunctionHelper.isExplicitNull(args, 2) ? parseRoundingMode(args[2]) : RoundingMode.HALF_UP;
-
     // An explicitly written null mode propagates, as every argument before it already does; only an omitted mode selects
-    // HALF_UP (issue #5629). This sits after the checks above so that round(null, 2, 'SIDEWAYS') still reports the
-    // unusable mode rather than answering null.
-    if (number == null || CypherFunctionHelper.isExplicitNull(args, 2))
+    // HALF_UP (issue #5629). It is not parsed on the way: parsing it only to discard the HALF_UP it would yield would
+    // also make parseRoundingMode's contract untrue. A mode that is present and not null is still parsed before
+    // propagation decides the answer, so round(null, 2, 'SIDEWAYS') reports the unusable mode rather than answering null.
+    final boolean nullMode = CypherFunctionHelper.isExplicitNull(args, 2);
+    final RoundingMode mode = args.length == 3 && !nullMode ? parseRoundingMode(args[2]) : RoundingMode.HALF_UP;
+
+    if (number == null || nullMode)
       return null;
 
     final double value = number.doubleValue();

